### PR TITLE
Adds a new example and adjusts some naming

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This adds a simpler example, which helps understand the basic ins and
outs. This will be used later when the %.wat parser is implemented.

Along the way, I noticed naming inconsistency. Most of the time,
function inputs are named params and outputs are named results. Using
the same vocab helps when mapping the spec to our code.

See https://www.w3.org/TR/wasm-core-1/#function-types%E2%91%A6

### CODE REVIEW

**Comment by mathetake** [comment]
> Along the way, I noticed naming inconsistency. Most of the time,
function inputs are named params and outputs are named results. Using
the same vocab helps when mapping the spec to our code.

excellent! 🚀

File: examples/simple_test.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	buf, err := os.ReadFile("wasm/simple.wasm")
+	require.NoError(t, err)
+	mod, err := wasm.DecodeModule(buf)
+	require.NoError(t, err)
+
+	// Create a new store and add the function "hello" which the module imports
+	store := wasm.NewStore(wazeroir.NewEngine())
+	hello := func(_ *wasm.HostFunctionCallContext) {
+		fmt.Println("Hello from Wazero!")
+	}
```
Comment (mathetake) [inline examples/simple_test.go:31]:
```suggestion
	var called bool
	hello := func(_ *wasm.HostFunctionCallContext) {
		called = false
	}
	
	// ...
	
	require.True(t, called)
```

**Review by mathetake — APPROVED** [review]

File: examples/simple_test.go (hunk 1 above)
Comment (mathetake) [reply examples/simple_test.go:31]: I meant I want to ensure the host function is actually called! :D

File: wasm/jit/engine.go [hunk 2]
```diff
@@ -108,11 +108,11 @@ func (e *engine) Call(f *wasm.FunctionInstance, args ...uint64) (returns []uint6
 		err = fmt.Errorf("function not compiled")
 		return
 	}
-	// Note the top value is the tail of the returns,
-	// so we assign the returns in reverse order.
-	returns = make([]uint64, len(f.Signature.ReturnTypes))
-	for i := range returns {
-		returns[len(returns)-1-i] = e.pop()
+	// Note the top value is the tail of the results,
+	// so we assign the results in reverse order.
```
Comment (codefromthecrypt) [inline wasm/jit/engine.go:112]:
```suggestion
	// so we assign them in reverse order.
```

File: wasm/store.go [hunk 3]
```diff
@@ -486,6 +481,18 @@ func (s *Store) buildFunctionInstances(module *Module, target *ModuleInstance) (
 	return rollbackFuncs, nil
 }
 
+// getFunctionName gets the name of the function corresponding to the codeIndex.
+func getFunctionName(functionNames map[uint32]string, importedFunctionCount int, codeIndex int) string {
```
Comment (codefromthecrypt) [inline wasm/store.go:485]: I didn't verify this with the compiler flag, but I assume it is inlineable

File: examples/simple_test.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+	"github.com/tetratelabs/wazero/wasm"
+	"github.com/tetratelabs/wazero/wasm/wazeroir"
+)
+
+// Test_Simple implements a basic function in go ("hello"), called by a function defined Web Assembly ("run"):
+func Test_Simple(t *testing.T) {
+	// Decode simple.wasm which was pre-compiled from this text format:
+	//	(module
+	//		(import "" "hello" (func $hello))
```
Comment (mathetake) [inline examples/simple_test.go:19]: I didn't know that imports with empty module name is valid 😄

File: examples/simple_test.go (hunk 1 above)
Comment (codefromthecrypt) [reply examples/simple_test.go:31]: ah good point! right now it is only a visual cue.

File: examples/simple_test.go (hunk 4 above)
Comment (codefromthecrypt) [reply examples/simple_test.go:19]: nutty right? it is all over the wasm spec tests!

**Comment by codefromthecrypt** [comment]
I switched the assertion to be more hello

**Review by mathetake — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
Thanks for the help @mathetake 🙇

**Pull request merged**